unison-core/src/Unison/Var.hs

```diff
@@ -87,6 +87,7 @@ rawName typ = case typ of
   Irrelevant -> "_irrelevant"
   UnnamedReference ref -> Reference.idToText ref
   UnnamedWatch k guid -> fromString k <> "." <> guid
+  Delay -> "()"
```

Comment:
Is this weird

Reply:
It's what we did before, though it happened during parsing - anytime we parsed a thunk, we'd parse it as a lambda with a variable called `()`.

I think ideally we'd just store the variable type in the database, but right now we convert everything to strings and this PR doesn't change that. So we need to pick a name that the user can't use and that the pretty-printer can reverse to decide when to use the thunk / `do` syntax.